docs/DeveloperGuide.md

```diff
@@ -660,3 +667,114 @@ testers are expected to do more *exploratory* testing.
    1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
 
 1. _{ more test cases …​ }_
+
+
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Planned Enhancements**
+
+### Do checks to ensure that old data is not the same as new data when editing data.
+
+Currently, Jobby sometimes allow editing of data such that the old data to be replaced with has the same contents as the new data.
+
+For example, `edit --application 1 --title SWE` on a job application with title "SWE" works, even though nothing is effectively changed.
+
+This can be done with a simple fix. The execute method for the editing of job applications is as follows:
+
+```java
+    public CommandResult execute(Model model) throws CommandException {
+        if (!editApplicationDescriptor.isAnyFieldEdited()) {
+            throw new CommandException(MESSAGE_NOT_EDITED);
+        }
+        List<JobApplication> lastShownList = model.getDisplayedApplicationList();
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPLICATION_DISPLAYED_INDEX);
+        }
+
+        JobApplication jobApplication = lastShownList.get(targetIndex.getZeroBased());
+        JobApplication newApplication = createApplication(jobApplication, editApplicationDescriptor);
+
+        try {
+            model.replaceApplication(targetIndex, newApplication);
+        } catch (IllegalValueException e) {
+            throw new CommandException(e.getMessage());
+        }
+
+        return new CommandResult(String.format(MESSAGE_EDIT_APPLICATION_SUCCESS, newApplication));
+    }
+```
+
+We can easily add a new line to check if the all the contents of the new application is the same as the old one, and throw a CommandException if it is.
+
+This can also be easily done for editing contacts.
+
+### Make commands only take in arguments that are applicable to them and reject other extra arguments.
+
+Currently, some commands with multiple modes share parameters between modes.
+
+Example: The edit command has a contact mode and job application mode. In contact mode, `--name` is used to edit the name of the contact. However, there is no such thing for job application mode.
+
+Yet, `edit --application 1 --name John --title SWE` works as long as a valid field that is used for job application is used, and ignores the extra arguments.
+
+This can be done with a simple fix. Every command alreadly has a list of flags that are accepted. 
+
+At the command parsing level, add additional checks against the list of flags provided by the command to ensure that every flag present in the command is applicable to the command used.
+
+<img src="images/enhancements/FlagChecker.png">
```

Comment:
We may consider mentioning that we have already implemented methods to do so but it was unused due to an oversight:

`argumentMultiMap.verifyNoExtraneousFlagsOnTopOf(Flag...)`